<a href="https://colab.research.google.com/github/aparna-2001/machine_learning_practice/blob/main/Diabetes_regression_model.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [16]:
import pandas as pd
import numpy as np
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
from sklearn.datasets import load_diabetes

In [24]:

data = load_diabetes()

diabetes_data = pd.DataFrame(data.data, columns=data.feature_names)
diabetes_data['target'] = data.target
diabetes_data.head()


age       sex       bmi        bp        s1        s2        s3  \
0  0.038076  0.050680  0.061696  0.021872 -0.044223 -0.034821 -0.043401   
1 -0.001882 -0.044642 -0.051474 -0.026328 -0.008449 -0.019163  0.074412   
2  0.085299  0.050680  0.044451 -0.005670 -0.045599 -0.034194 -0.032356   
3 -0.089063 -0.044642 -0.011595 -0.036656  0.012191  0.024991 -0.036038   
4  0.005383 -0.044642 -0.036385  0.021872  0.003935  0.015596  0.008142   

         s4        s5        s6  target  
0 -0.002592  0.019907 -0.017646   151.0  
1 -0.039493 -0.068332 -0.092204    75.0  
2 -0.002592  0.002861 -0.025930   141.0  
3  0.034309  0.022688 -0.009362   206.0  
4 -0.002592 -0.031988 -0.046641   135.0

In [19]:
diabetes_data.shape

(442, 10)

In [21]:
list(diabetes_data.columns)

['age', 'sex', 'bmi', 'bp', 's1', 's2', 's3', 's4', 's5', 's6']

In [22]:
diabetes_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 442 entries, 0 to 441
Data columns (total 10 columns):
 #   Column  Non-Null Count  Dtype  
---  ------  --------------  -----  
 0   age     442 non-null    float64
 1   sex     442 non-null    float64
 2   bmi     442 non-null    float64
 3   bp      442 non-null    float64
 4   s1      442 non-null    float64
 5   s2      442 non-null    float64
 6   s3      442 non-null    float64
 7   s4      442 non-null    float64
 8   s5      442 non-null    float64
 9   s6      442 non-null    float64
dtypes: float64(10)
memory usage: 34.7 KB




*   The data contains 442 rows and 10 columns.

*   All the data is in float.

* All columns contain 442 values, so there are no null values.





*   Modeling regression between blood pressure and target (diseases progression score after one year)

*   Independent variable, x = blood pressure

*   Dependent variable, y = target





In [26]:
#adding constant to x values, so the y line not pass through the origin

x = sm.add_constant(diabetes_data['bp'])
y = diabetes_data['target']
x.head()

const        bp
0    1.0  0.021872
1    1.0 -0.026328
2    1.0 -0.005670
3    1.0 -0.036656
4    1.0  0.021872

In [27]:
#training and testing data

train_x, test_x, train_y, test_y = train_test_split(x,
                                                    y,
                                                    train_size = 0.8,
                                                    random_state = 100)

In [30]:
#fitting the model

diabetes_model = sm.OLS(train_y, train_x).fit()
print(diabetes_model.params)

const    153.041444
bp       675.853574
dtype: float64




*   Estimated model can be written as in the form of straight line;

    y = mx + c ;

    target = 675.853574*bp + 153.041444



